src/main/java/io/appium/java_client/android/AndroidMobileOptions.java

```diff
@@ -0,0 +1,1716 @@
+        return (boolean) getCapability(AndroidMobileCapabilityType.CHROMEDRIVER_DISABLE_BUILD_CHECK);
+    }
+
+    /**
+     * Set the path to webdriver executable.
+     *
+     * @param path is the absolute local path to webdriver executable.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#CHROMEDRIVER_EXECUTABLE
+     */
+    public AndroidMobileOptions setChromedriverExecutable(String path) {
+        return amend(AndroidMobileCapabilityType.CHROMEDRIVER_EXECUTABLE, path);
+    }
+
+    /**
+     * Get the path to webdriver executable.
+     *
+     * @return String of the path to webdriver executable.
+     * @see AndroidMobileCapabilityType#CHROMEDRIVER_EXECUTABLE
+     */
+    public String getChromedriverExecutable() {
+        return (String) getCapability(AndroidMobileCapabilityType.CHROMEDRIVER_EXECUTABLE);
+    }
+
+    /**
+     * Set the directory to look for Chromedriver executables in, for automatic discovery of compatible Chromedrivers.
+     *
+     * @param dir is the absolute path of the directory to automatically discover compatible Chromedriver executable.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#CHROMEDRIVER_EXECUTABLE_DIR
+     */
+    public AndroidMobileOptions setChromedriverExecutableDir(String dir) {
+        return amend(AndroidMobileCapabilityType.CHROMEDRIVER_EXECUTABLE_DIR, dir);
+    }
+
+    /**
+     * Get the directory to look for Chromedriver executables in, for automatic discovery of compatible Chromedrivers.
+     *
+     * @return String of the directory of Chromedriver executables.
+     * @see AndroidMobileCapabilityType#CHROMEDRIVER_EXECUTABLE_DIR
+     */
+    public String getChromedriverExecutableDir() {
+        return (String) getCapability(AndroidMobileCapabilityType.CHROMEDRIVER_EXECUTABLE_DIR);
+    }
+
+    /**
+     * Set the port to start Chromedriver on.
+     *
+     * @param port is the number of the port to use for Chromedriver.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#CHROMEDRIVER_PORT
+     */
+    public AndroidMobileOptions setChromedriverPort(int port) {
+        return amend(AndroidMobileCapabilityType.CHROMEDRIVER_PORT, port);
+    }
+
+    /**
+     * Get the port Chromedriver starts on.
+     *
+     * @return Integer of the port Chromedriver starts on.
+     * @see AndroidMobileCapabilityType#CHROMEDRIVER_PORT
+     */
+    public int getChromedriverPort() {
+        return (int) getCapability(AndroidMobileCapabilityType.CHROMEDRIVER_PORT);
+    }
+
+    /**
+     * Set the valid ports for Appium to use for communication with Chromedrivers.
+     *
+     * @param ports is the list of ports for Appium to use with Chromedrivers.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#BUILD_TOOLS_VERSION
+     */
+    public AndroidMobileOptions setChromedriverPorts(List<Integer> ports) {
```

Comment:
is it really a list?

Reply:
Yes, [according to the docs](https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md#android-only) it is a list and not a comma separated string.
Though, this example would indicate that it isn't only a List of Integers:
> e.g. `[8000, [9000, 9005]]`
c5fbb3dc67a0891a002bbca235595fe13a8c3351

Reply:
https://github.com/appium/appium-android-driver/blob/e2a7e4b5407b845ce7f21b8e4b0be0c2cc0fd01a/lib/desired-caps.js#L91